### PR TITLE
Update to Application Insights 2.3.0

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/Engineering/issues/2527.
Depends on https://github.com/NuGet/ServerCommon/pull/302.

This allows us to avoid https://github.com/microsoft/ApplicationInsights-dotnet/issues/480.

### CODE REVIEW

**Comment by joelverhagen** [comment]
This PR is not a hotfix. I started with this approach since there are a couple changes in DEV around dependencies and I didn't want a painful merge later. Also, my confidence is high from load testing.

**Comment by joelverhagen** [comment]
> Also, my confidence is high from load testing.

Fateful words 😓

File: src/NuGet.Services.Entities/NuGet.Services.Entities.csproj [hunk 1]
```diff
@@ -100,7 +100,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="Newtonsoft.Json">
-      <Version>9.0.1</Version>
+      <Version>10.0.2</Version>
```
Comment (cristinamanum) [inline src/NuGet.Services.Entities/NuGet.Services.Entities.csproj]: Can we take this upgrade later?

Comment (joelverhagen) [reply src/NuGet.Services.Entities/NuGet.Services.Entities.csproj]: This is required for ServerCommon upgrade.

File: src/AccountDeleter/AccountDeleter.csproj [hunk 2]
```diff
@@ -103,16 +103,16 @@
       <Version>4.1.0-master-2602271</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Configuration">
-      <Version>2.50.0</Version>
+      <Version>2.52.0-jver-ai-23-2799726</Version>
```
Comment (ryuyu) [inline src/AccountDeleter/AccountDeleter.csproj]: Is it possible for us to take a release of server common from master? If not, this is fine, but i would prefer for us not to depend on dev builds.

Comment (joelverhagen) [reply src/AccountDeleter/AccountDeleter.csproj]: This is covered by https://github.com/NuGet/ServerCommon/pull/302 and will be pulled into this PR prior to merge.

**Comment by joelverhagen** [comment]
Closing in lieu of https://github.com/NuGet/NuGetGallery/pull/7309. I had a weekend to ponder 💤

**Pull request closed without merging**